### ISSUE TITLE
What is the best way to emulate PostgreSQL serial/indentity (auto increment ID column) in Questdb ?

### ISSUE THREAD

**Issue by lkpatel** [questdb/questdb#2869]
### Describe the bug

How to add auto increment column (like serial data type in PostgreSQL) in Questdb ? This is not for synthetic data generation.

### Environment

```markdown
- **QuestDB version**:
- **OS**:
- **Browser**:
```

**Comment by puzpuzpuz** [comment]
@jerrinot is working on UUID support which will include UUID v4 generation function. It could be used to produce and store unique identifiers for reference data tables.

**Comment by glasstiger** [comment]
It is not the most efficient way to do it and also depends on how you ingest the data but you might be able to use the `max()` function as a workaround.

Small example:
```
create table currencies(id long, ccy symbol, ts timestamp) timestamp(ts);
insert into currencies values (1, 'USD', '2019-03-10T00:00:00.000000Z');
insert into currencies select max(id) + 1, 'EUR', '2019-03-10T01:00:00.000000Z' from currencies;
insert into currencies select max(id) + 1, 'GBP', '2019-03-10T02:00:00.000000Z' from currencies;
```

**Comment by lkpatel** [comment]
Is max() function thread safe ?

**Comment by glasstiger** [comment]
yes, when `INSERT`, `UPDATE` or `ALTER TABLE` runs the table is locked for write.
if you always use `max()` to get the next id you will get a monotonously increasing sequence of unique ids.

**Comment by puzpuzpuz** [comment]
Client-side races are still possible with `max()`. Multiple threads/connections can read the same `max()` value and then insert the next value one after each other due to thread scheduling delays. Such scenario would lead to duplicate values inserted into the table.

**Comment by glasstiger** [comment]
Fair enough, what I meant is that `max()` combined with `insert as select` is safe because `INSERT` locks the table. 
If you always use `insert into <table> select max(id) + 1, <some other fields> from <table>` to insert data you will get a monotonously increasing sequence of unique ids.
This should work from multiple connections/threads too.

If you asked about running `select max(id) as maxId from <table>` and then `insert into <table> values (maxId, <other fields>)` that will not work.
